Source/DiabloUI/hero/selhero.cpp

```diff
@@ -169,22 +169,31 @@ void SelheroListSelect(size_t value)
 
 		vecSelHeroDlgItems.clear();
 		int itemH = 33;
-		vecSelHeroDlgItems.push_back(std::make_unique<UiListItem>(_("Warrior"), static_cast<int>(HeroClass::Warrior)));
-		vecSelHeroDlgItems.push_back(std::make_unique<UiListItem>(_("Rogue"), static_cast<int>(HeroClass::Rogue)));
-		vecSelHeroDlgItems.push_back(std::make_unique<UiListItem>(_("Sorcerer"), static_cast<int>(HeroClass::Sorcerer)));
-		if (gbIsHellfire) {
-			vecSelHeroDlgItems.push_back(std::make_unique<UiListItem>(_("Monk"), static_cast<int>(HeroClass::Monk)));
-			if (HaveBardAssets() || *GetOptions().Gameplay.testBard) {
-				vecSelHeroDlgItems.push_back(std::make_unique<UiListItem>(_("Bard"), static_cast<int>(HeroClass::Bard)));
+		for (size_t i = 0; i < GetNumPlayerClasses(); ++i) {
+			const HeroClass heroClass = static_cast<HeroClass>(i);
+
+			if ((heroClass == HeroClass::Monk || heroClass == HeroClass::Bard || heroClass == HeroClass::Barbarian) && !gbIsHellfire) {

```

Comment:
We have historically considered the two extra classes as part of the engine and not Hellfire specifically
```suggestion
			if (heroClass == HeroClass::Monk && !gbIsHellfire) {
```